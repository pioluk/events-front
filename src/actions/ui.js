export function selectEvent(color, image) {
  return {
    type: 'SELECT_EVENT',
    color,
    image
  }
}
