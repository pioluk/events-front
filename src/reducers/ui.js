/* @flow */

type UiState = {
  selectedEventColor: ?string,
  selectedEventImage: ?string
}

const initialState: UiState = {
  selectedEventColor: null,
  selectedEventImage: null
}

export default function uiReducer (state: UiState = initialState, action: any): UiState {
  switch (action.type) {
    case 'SELECT_EVENT':
      return {
        selectedEventColor: action.color,
        selectedEventImage: action.image
      }

    default:
      return state
  }
}
