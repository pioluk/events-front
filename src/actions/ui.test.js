// global describe,it,expect

import * as actions from './ui'

describe('ui actions', () => {
  it('should create an action to select an event', () => {
    const color = '#a5aaf2'
    const image = 'ec8672d6-3787-4ce6-a35b-715d012f23e4.jpg'
    expect(actions.selectEvent(color, image)).toMatchSnapshot()
  })
})
