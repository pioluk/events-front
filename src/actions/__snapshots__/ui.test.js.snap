exports[`ui actions should create an action to select an event 1`] = `
Object {
  "color": "#a5aaf2",
  "image": "ec8672d6-3787-4ce6-a35b-715d012f23e4.jpg",
  "type": "SELECT_EVENT",
}
`;
